lib/ramble/ramble/util/logger.py

```diff
@@ -0,0 +1,201 @@
+
+    def _stream_kwargs(self, default_kwargs={}, index=None):
+        """Construct keyword arguments for a stream
+
+        Build keyword arguments of the form: {'stream': <log_stream>} to allow
+        LLNL's tty utility to print to a specific stream.
+
+        When default_kwargs are passed in, these are applied on top of the
+        constructed kwargs.
+
+        When index is passed in, the stream added into kwargs is the log in
+        position index within the stack (where -1 is considered active).
+
+        Args:
+            default_kwargs: Default keyword arguments to use
+            index: Index of log (in stack) to build kwargs for
+
+        Returns:
+            kwargs: Constructed kwargs with defaults applied
+        """
+        kwargs = {}
+        stream_index = None
+        if index is not None:
+            if index >= 0 and index <= len(self.log_stack):
+                stream_index = index
+            else:
+                tty.die(
+                    f'Error: Requested stream index of {index} is outside of '
+                    f'the stream range of 0 - {len(self.log_stack)}'
+                )
+
+        else:
+            if len(self.log_stack) > 0:
+                stream_index = len(self.log_stack) - 1
+
+        if stream_index is not None:
+            kwargs['stream'] = self.log_stack[stream_index][1]
+
+        kwargs.update(default_kwargs)
+
+        return kwargs
+
+    def all_msg(self, *args, **kwargs):
+        """Print a message to all logs
+
+        Pass all args and kwargs to tty.info (which will concatenate and
+        print). Perform this action for all logs and the default log (to
+        screen).
+        """
+        for idx, log in enumerate(self.log_stack):
+            st_kwargs = self._stream_kwargs(default_kwargs=kwargs, index=idx)
+            tty.info(*args, **st_kwargs)
+
+        tty.info(*args, **kwargs)
+
+    def msg(self, *args, **kwargs):
+        """Print a message to the active log
+
+        Pass all args and kwargs to tty.info (which will concatenate and
+        print). Perform this action for the active log only.
+        """
+        st_kwargs = self._stream_kwargs(default_kwargs=kwargs)
+        tty.info(*args, **st_kwargs)
+
+    def info(self, *args, **kwargs):
+        """Print a message to the active log
+
+        Pass all args and kwargs to tty.info (which will concatenate and
+        print). Perform this action for the active log only.
+        """
+        st_kwargs = self._stream_kwargs(default_kwargs=kwargs)
+        tty.info(*args, **st_kwargs)
+
+    def verbose(self, *args, **kwargs):
+        """Print a verbose message to the active log
+
+        Pass all args and kwargs to tty.verbose (which will concatenate and
+        print). Perform this action for the active log only.
+        """
+        st_kwargs = self._stream_kwargs(default_kwargs=kwargs)
+        tty.verbose(*args, **st_kwargs)
+
+    def warn(self, *args, **kwargs):
```

Comment:
How do you feel about warnings going to all log files like errors do? I think warnings making it to the terminal is probably a good idea (or at the very least I think we want some sort of call with the semantic of 'send this to the screen too')

Reply:
I was trying to figure out a nice semantic for "print to screen" as well.

Maybe a kwarg? something like `tee=True`?  Maybe I'll just make warning print to the active stream and the CLI for the time being, and we can see how we like that.